### PR TITLE
Remove cutter win32

### PR DESCRIPTION
**Pull request by mrexodia**
To do:

- testing
- documentation
- make appveyor call `prepare_r2.bat` + use jom to compile `cutter.pro` instead of this shitty vcxproj generation
- radare2's `meson.bat` doesn't actually support XP (static libraries also need to be fixed)
- radare2's `meson.bat` doesn't actually support msbuild (it builds, but doesn't output anything in the destination folder)

Quick guide:

1. run `prepare_r2.bat` it will compile r2 and extract the relevant files in the `cutter_win32` directory (submodule was removed, but r2 doesn't like windows so it's required)
2. Open `cutter.pro` with Qt creator and compile it with msvc2015 qt 5.9 in release mode
3. Once compiled (quite magic, r2 is not suitable at all for dynamic linking), copy the contents of `dist32` in the `build-cutter-Qt-blahblah\release` so you have `cutter.exe` next to `radare.exe`
4. open cutter.exe and feel the hacks crumble to dust in your hands

There are tons of problems on windows (`share` folder not recognized whatsoever, sdb is fishy as fuck because it's statically compiled in multiple dlls, cutter crashes when restoring a previously-saved file, ui is bugged, fonts don't work, etc, etc).

PS `prepare_r2.bat` wasn't mean to be beautiful, but ninja/meson are kinda broken so git clean is required

### CODE REVIEW

File: cutter_win32/radare2/include/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Relevant files will be placed here after running prepare_r2.bat
+/libr/
```
Comment (xarkes) [inline cutter_win32/radare2/include/.gitignore]: What's the point to duplicate those files? Can't you just put one in cutter_win32 directory?

File: prepare_r2.bat [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+copy ..\ninja.exe .\
+copy ..\meson.py .\
+rmdir /s /q ..\dist32
+call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x86
+call meson.bat --release --shared
+call sys\meson_install.bat --with-static ..\dist32
+copy /Y build\r_userconf.h ..\dist32\include
+copy /Y build\r_version.h ..\dist32\include
+copy /Y build\shlr\liblibr2sdb.a ..\dist32\r_sdb.lib
+
```
Comment (xarkes) [inline prepare_r2.bat]: Imho it would be better to have an IF ELSE and add another job for x86 in appveyor

File: prepare_r2.bat [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+rmdir /s /q ..\dist64
+call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x64
+call meson.bat --release --shared
+call sys\meson_install.bat --with-static ..\dist64
+copy /Y build\shlr\liblibr2sdb.a ..\dist64\r_sdb.lib
+
+cd ..
+
+echo Copying relevant files in cutter_win32
+xcopy /Y dist32\include\libr cutter_win32\radare2\include\libr
```
Comment (xarkes) [inline prepare_r2.bat]: Why is this required? Can't you just use -Iradare\include\libr and -Ibuild ?

**Comment by xarkes** [comment]
And also I think it would be cleaner if you just created the cutter_win32 folder tree on runtime, I don't think we have to keep it into the repo.

File: cutter_win32/radare2/include/.gitignore (hunk 1 above)
Comment (mrexodia) [reply cutter_win32/radare2/include/.gitignore]: It's to keep the changes as minimal as possible. The `.gitignore` is to create an empty directory structure that users can notice if they mess up.

File: prepare_r2.bat (hunk 2 above)
Comment (mrexodia) [reply prepare_r2.bat]: This file is not meant for AppVeyor, it is meant for users and users don't want to run multiple files to initialize a repository. Also, compiling radare is separate from AppVeyor and I agree that there should be two separate jobs for compiling cutter on AppVeyor.

File: prepare_r2.bat [hunk 4]
```diff
@@ -39,9 +39,9 @@ copy /Y build\shlr\liblibr2sdb.a ..\dist64\r_sdb.lib
 cd ..
 
 echo Copying relevant files in cutter_win32
-xcopy /Y dist32\include\libr cutter_win32\radare2\include\libr
+xcopy /Y dist32\include\libr cutter_win32\radare2\include\libr\
```
Comment (mrexodia) [inline prepare_r2.bat]: This does not contribute to the build hanging and changes the meaning of the script.

File: prepare_r2.bat (hunk 3 above)
Comment (mrexodia) [reply prepare_r2.bat]: This is to `radare2.pri` will actually find the r2 include files.

**Comment by mrexodia** [comment]
We have to keep `cutter_win32` because there are various hacks there to make radare headers work with MSVC.

**Comment by xarkes** [comment]
Ack

**Comment by xarkes** [comment]
Thanks a lot!
